### DISCUSSION TITLE
How to use pinia in electron main process

### DISCUSSION THREAD

**Discussion by bayun2** [vuejs/pinia#653] [Help and Questions] [answered]
I want to write a  plugin like https://www.npmjs.com/package/@kasvith/electron-vuex
But I have two problem.
1.  Pinia has't mutation, so I can't notify renderer process to modify the specified  state when main process state change like https://github.com/vue-electron/vuex-electron/blob/master/src/shared-mutations.js#L91.
2. In main process,if I do like below it will warning, beacause there is no window

```
const pinia = createPinia()
const app = createApp({})
app.use(pinia)
```

**Comment by AntonDeMagic** [comment, marked as answer]
Yes, this is an old post, but we still find it while we're looking for a solution.
We had the same struggle with our Electron app and finally decided to build a [plugin](https://www.npmjs.com/package/electron-pinia-sync) to keep stores in sync between processes without the IPC headache. Feel free to give it a spin and let us know what you think!

Hope it helps!
